debug_module.tex

```diff
@@ -177,24 +177,24 @@ \section{Hart States}
 \section{Run Control} \label{runcontrol}
 
 For every hart, the Debug Module contains 4 conceptual bits of state: halt
-request, resume request, halt-on-reset request,  and hart reset.
+request, resume ack, halt-on-reset request,  and hart reset.
 (The hart reset and halt-on-reset request bits are optional.)
 These bits all reset to 0. A debugger can write them for the currently selected
-harts through \Fhaltreq, \Fresumereq, \Fsetresethaltreq/\Fclrresethaltreq and
-\Fhartreset in \Rdmcontrol, only one of which may be high in any single write.
-In addition the DM receives halted, running, and resume ack signals from each
-hart.
+harts through \Fhaltreq, \Fsetresethaltreq/\Fclrresethaltreq and
+\Fhartreset in \Rdmcontrol.
```

Comment:
We aren't explaining here how resume ack gets set, so it seems like a weird omission from this list. It is explained later, but shouldn't we say something like "A debugger can change their value for the currently selected hart through... \Fresumereq ... " 

I think it's weird to say resume ack is the bit of state. It's not like a hart sees the conceptual resume ack go low and then does something. Harts see the resume request go low, and then resume, and Debug module records that they resumed in resume ack

Reply:
Ok, fine if you want to track resumeack instead of resumereq. But this paragraph is weird from a readability standpoint.  This paragraph starts with a list of bits and it seems like it's about how to modify those bits. But you don't in this paragraph actually say anything about resume ack bit. Yes, the debugger doesn't directly "set" resume ack, but it does modify it:  it can be cleared by a write to \Fresumereq as you describe below. Does it need a forward reference?

Reply:
I'm tempted to make this paragraph simply:
For every hart, the Debug Module contains 4 conceptual bits of state: halt
request, resume ack, halt-on-reset request,  and hart reset.
(The hart reset and halt-on-reset request bits are optional.)
These bits all reset to 0. ~~A debugger can write them for the currently selected
harts through \Fhaltreq, \Fsetresethaltreq/\Fclrresethaltreq and
\Fhartreset in \Rdmcontrol.~~
In addition the DM receives halted and running signals from each hart.

Then update the following paragraphs with a sentence on which bit the debugger writes to get a conceptual bit/signal to do something.

Thoughts?

Reply:
Yeah

Reply:
Also maybe “contains” should be “tracks” or “maintains”?

Reply:
Changed to tracks.